### ISSUE TITLE
npm install does't update dependencies in devDependencies if change their versions in package json?

### ISSUE THREAD

**Issue by klimashkin** [npm/npm#8032]
In 2.8.4

**Comment by th0r** [comment]
+1 `npm 2.8.3`, `node 0.10.38`.
Steps to reproduce:
 1. Make `package.json` with this content:
```json
{
  "name": "npm-test",
  "version": "1.0.0",
  "devDependencies": {
    "webpack": "1.8.7"
  }
}
```
2. Exec `npm i`. Webpack `1.8.7` will be installed into `node_modules`.
3. Change `1.8.7` to `1.8.8` in `package.json` and save it.
4. Exec `npm i`. Nothing happens, webpack is still `1.8.7`.

**Comment by othiym23** [comment]
This should be fixed by 9ad210042242e51d52b2a8b633d8e59248f5faa4, and was caused by a regression introduced by a change to `read-package-json`. Sorry about that!
